app/models/api/v3/node_type.rb

```diff
@@ -3,6 +3,30 @@ module V3
     class NodeType < BaseModel
       has_many :context_node_types
       has_many :nodes
+
+      PLACES = [
+        NodeTypeName::MUNICIPALITY,
+        NodeTypeName::LOGISTICS_HUB,
+        NodeTypeName::BIOME,
+        NodeTypeName::STATE
+      ].freeze
+
+      def self.node_index_for_name(context, node_type_name)
+        zero_based_idx = ContextNode.
+          joins(:node_type).
+          where(context_id: context.id).
+          where('node_types.node_type' => node_type_name).
+          pluck(:column_position).first
+        zero_based_idx && zero_based_idx + 1 || 0
```

Comment:
Isn't this the same as `zero_based_idx + 1 || 0`?

Reply:
I think `zero_based_idx` can be nil, in which case `+` would result in "undefined method + for nil"